Sources/IkigaJSON/Codable/JSONDecoder.swift

```diff
@@ -132,10 +132,7 @@ public final class IkigaJSONDecoder {
     /// Parses the Decodable type from `[UInt8]`. This is the equivalent for JSONDecoder's Decode function.
     public func decode<D: Decodable>(_ type: D.Type, from string: String) throws -> D {
         // TODO: Optimize with Swift 5
-        guard let data = string.data(using: .utf8) else {
-            throw JSONParserError.invalidData(string)
-        }
-        
+        let data = string.data(using: .utf8)!
         return try self.decode(type, from: data)
     }
```

Comment:
I suggest this implementation, with that `TODO` in mind and knowing that `.data(using:)` is just about the most inefficient way to convert a `String` to UTF-8 that you can get without deliberately invoking `NSString`.

These implementations **REPLACE** the existing `[UInt8]` and `String` overloads (and in fully backwards-source-compatible fashion, for style points!).

```swift
// Using `[UInt8]` as the input means the method only tolerates `Array` _specifically_, but in reality most `Sequence`s can be efficiently accessed as unsafe buffers (especially `StringProtocol.UTF8View`, for example)
public func decode<D: Decodable, S: Sequence>(_: D.Type, from bytes: S) throws -> D where S.Element == UInt8 {
    // N.B.: As of Swift 5.3, everything String-related and almost every sequence and collection type (other than transforms like LazySequence) provides contiguous storage. This is currently the most efficient known way of getting UTF-8 bytes from a non-bridged String.
    return try bytes.withContiguousStorageIfAvailable {
        try self.decode(D.self, from: $0)
    } ?? self.decode(D.self, from: Data(bytes))
}

// Use `StringProtocol` so both `String` and `Substring` "just work" with correct semantics.
public func decode<D: Decodable, S: StringProtocol>(_: D.Type, from string: S) throws -> D {
    // `.withUTF8()` is mutating, so can't use it here, use UTF8View to go through the Sequence overload
    return try self.decode(D.self, from: string.utf8)
}
```

P.S.: I did not check to see if this actually compiles or works. Written in GH's Web interface, quite possibly (even likely) buggy.